distage/distage-core-api/src/main/scala/izumi/distage/model/definition/ModuleBase.scala

```diff
@@ -59,6 +59,15 @@ object ModuleBase {
       T.make(module.bindings.map(f))
     }
 
+    def foldWith[B](b: B)(f: (Binding, B) => (Binding, B)): (T, B) = {
```

Comment:
I've add here new method, to log strengthen bindings without using mutable collections.

Reply:
Might as well add a more general `fold[B](f: (B, Binding) => B): B` and base this on it. Also I guess you need to specify the direction of the accumulator parameter, because

1. A Set fold may not actually be unordered, because there's `ListSet` & `LinkedHashSet` - ordered Sets in stdlib
2. There are still two variants of where to place the accumulator parameter in the stdlib. – currently the accumulator parameter is to the right, as in `foldRight`, so I guess this is `foldRightWith` ?